astronomer/providers/sftp/hooks/sftp.py

```diff
@@ -78,113 +80,126 @@ async def _get_conn(self) -> asyncssh.SSHClientConnection:
-                        "hostname": conn.host,
-                        "username": conn.login,
-                        "password": conn.password,
-                        "port": conn.port,
-                    }
-
-                    if self.private_key:
-                        connect_kwargs.update(pkey=self.private_key)
-
-                    if self.key_file:
-                        connect_kwargs.update(key_filename=self.key_file)
-
-                    def log_before_sleep(retry_state):
-                        return self.log.info(
-                            "Failed to connect. Sleeping before retry attempt %d", retry_state.attempt_number
-                        )
-
-                    for attempt in Retrying(
-                        reraise=True,
-                        wait=wait_fixed(3) + wait_random(0, 2),
-                        stop=stop_after_attempt(3),
-                        before_sleep=log_before_sleep,
-                    ):
-                        with attempt:
-                            paramiko_client.connect(**connect_kwargs)
-
-            conn_config = {
-                "host": conn.host,
-                "port": conn.port,
+        conn = await sync_to_async(self.get_connection)(self.sftp_conn_id)
+        if conn.extra is not None:
+            extra_options = conn.extra_dejson
+            if "key_file" in extra_options and self.key_file == "":
+                self.key_file = extra_options.get("key_file")
+            if "known_hosts" in extra_options:
+                self.known_hosts = extra_options.get("known_hosts")
+            if ("passphrase" or "private_key_passphrase") in extra_options:
+                self.passphrase = extra_options.get("passphrase")
+            if "private_key" in extra_options:
+                self.private_key = extra_options.get("private_key")
+
+            host_key = extra_options.get("host_key")
+            no_host_key_check = extra_options.get("no_host_key_check")
+
+            if no_host_key_check is not None:
+                no_host_key_check = str(no_host_key_check).lower() == "true"
+                if host_key is not None and no_host_key_check:
+                    raise ValueError("Must check host key when provided.")
+                self.no_host_key_check = no_host_key_check
+
+            if host_key is not None:
+                if host_key.startswith("ssh-"):
+                    key_type, host_key = host_key.split(None)[:2]
+                    key_constructor = self._host_key_mappings[key_type[4:]]
+                elif host_key.startswith("ecdsa-"):
+                    _, host_key = host_key.split(None)[:2]
+                    key_constructor = paramiko.ECDSAKey
+                else:
+                    key_constructor = paramiko.RSAKey
+                decoded_host_key = decodebytes(host_key.encode("utf-8"))
+                self.host_key = key_constructor(data=decoded_host_key)
+
+        if self.no_host_key_check:
```

Comment:
I feel rather than having multiple if else at same place we should split it into corresponding methods.